java/parser/test/ParserTest.java

```diff
@@ -408,13 +410,109 @@ public void testLongPredicateQuery() {
         assertQueryEquals(expected, parsed, query);
     }
 
+    @Test
+    public void testAttributeQueryByValueVariable() {
+        final String query = "match\n" +
+                "?x = 5;\n" +
+                "$a == ?x isa age;";
+        TypeQLMatch parsed = TypeQL.parseQuery(query).asMatch();
+        TypeQLMatch expected = match(
+                valueVar("x").assign(Expression.constant(5)),
+                var("a").eq(valueVar("x")).isa("age"));
+        assertQueryEquals(expected, parsed, query);
+    }
+
+    @Test
+    public void testAssignOps() {
+        final String query = "match\n" +
+                "$x isa commodity,\n" +
+                "    has price $p;\n" +
+                "(commodity: $x, qty: $q) isa order;\n" +
+                "?net = $p * $q;\n" +
+                "?gross = ?net * ( 1.0 + 0.21 );";
+        TypeQLMatch parsed = TypeQL.parseQuery(query).asMatch();
+        TypeQLMatch expected = match(
+                var("x").isa("commodity").has("price", var("p")),
+                rel("commodity", "x").rel("qty", "q").isa("order"),
+                valueVar("net").assign(Expression.times(Expression.thingVar(var("p")), Expression.thingVar(var("q")))),
+                valueVar("gross").assign(Expression.times(
+                        Expression.valueVar(valueVar("net")),
```

Comment:
let's look further to see if we can get rid of the explicit `Expression` wrapper around a variable of either value or concept variants

Reply:
Made both implement `ExpressionBuilder<T extends Expression>` with a method `T toExpression`
This lets us write `Expression.plus(var("x"), valueVar("y"))`